### PR TITLE
provide empty 'dm' object

### PR DESCRIPTION
**Pull request by TSchiefer**
- Empty object can be created via `dm()` and `new_dm()` (without arguments, similar to `numeric()`, `list()`, etc.)
- Allows for creation of a `dm` from scratch, adding tables successively with `cdm_add_tbl()`
- Works locally and on DB
- vignettes have NOT been adapted

### CODE REVIEW

File: R/select-tbl.R [hunk 1]
```diff
@@ -49,6 +49,8 @@ tidyselect_table_names <- function(dm) {
 }
 
 cdm_select_tbl_impl <- function(dm, selected) {
+
+  if (is_empty(selected)) return(empty_dm())
```
Comment (krlmlr) [inline R/select-tbl.R]: Do we really need the shortcut here?

Comment (TSchiefer) [reply R/select-tbl.R]: We get an error otherwise:
```r
Error in dot_call(capture_dots, frame_env = frame_env, named = named,  : 
  attempt to set an attribute on NULL
```

Comment (krlmlr) [reply R/select-tbl.R]: Where does that error occur? Also with `selected = character()`?

Comment (krlmlr) [inline R/select-tbl.R]:
```suggestion
  # Required to avoid error further below
  if (is_empty(selected)) return(empty_dm())
```

Comment (TSchiefer) [reply R/select-tbl.R]: in `filter_recode_table(selected)` when `selected` is `named character(0)`

File: R/dm.R [hunk 2]
```diff
@@ -4,7 +4,7 @@
 #' The `dm` class wraps [dplyr::src] and adds a description of table relationships
 #' based on [datamodelr::datamodelr-package].
 #'
-#' `dm()` coerces its inputs.
+#' `dm()` coerces its inputs. If called without parameters, an empty `dm` object is created.
```
Comment (krlmlr) [inline R/dm.R]:
```suggestion
#' `dm()` coerces its inputs. If called without arguments, an empty `dm` object is created.
```

File: R/dm.R [hunk 3]
```diff
@@ -56,11 +57,14 @@ dm <- nse_function(c(src, data_model = NULL), ~ {
 #' Low-level constructor
 #'
 #' `new_dm()` only checks if the inputs are of the correct class.
+#' If called without parameters, an empty `dm` object is created.
```
Comment (krlmlr) [inline R/dm.R]:
```suggestion
#' If called without arguments, an empty `dm` object is created.
```

**Comment by krlmlr** [comment]
Thanks! Need to add a test that `cdm_select_tbl(dm)` is empty.
